react/Paywall/Paywall.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { FC } from 'react'
+
+import Icon from '../Icon'
+
+import { IllustrationDialog } from '../CozyDialogs'
+import Button from '../Button'
+import CozyUpgrade2Icon from '../Icons/CozyUpgrade2'
+import { Typography } from '@material-ui/core'
+
+interface PaywallProps {
+  icon?: JSX.Element
+  title: string
+  content: JSX.Element | string
+  actionLabel: string
+  onClose: () => void
+  onAction: () => void
+}
+
+const Paywall: FC<PaywallProps> = ({
+  icon = CozyUpgrade2Icon,
+  title,
+  content,
+
+  actionLabel,
+  onClose,
+  onAction
+}) => (
+  <IllustrationDialog
+    size="small"
+    actionsLayout="column"
+    open={true}
+    title={<Icon icon={icon} width={128} height={128} />}
+    content={
+      <div className="u-flex u-flex-column u-flex-items-center u-ta-center">
+        <Typography variant="h3" className="u-mb-1">
+          {title}
+        </Typography>
+        <div>{content}</div>
+      </div>
+    }
+    actions={<Button theme="primary" label={actionLabel} onClick={onAction} />}
+    onClose={onClose}
+  />
+)
+
+export default Paywall
```

Comment:
comme discuté de vive voix, est-ce qu'on ne mettrait pas ça dans le dossier Dialog ?

Reply:
I was staying on a paywall folder with the different specific versions (ex: OnlyOffice, ...) like CozyDialog